subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -105,16 +170,46 @@ int dfu_target_mcuboot_write(const void *const buf, size_t len)
 		return err;
 	}
 
-	offset += len;
+	err = store_flash_img_context();
```

Comment:
Is each call to `dfu_target_mcuboot_write` going to write to flash? If `len` can be arbitrary, we might do too many writes in certain situations. It would be nice to have a config to let users decide after how many bytes store the progress, e.g. every 64, 512, 1024 etc.

Reply:
I would say that this could be a "nice to have" which we can add if deemed necessary. It is not trivial, since you don't know when the flash driver will actually write the buffered data to flash. This state is inside the flash context being stored.